docs/intro/arch_overview/dynamic_configuration.rst

```diff
@@ -0,0 +1,66 @@
+.. _arch_overview_dynamic_config:
+
+Dynamic configuration
+=====================
+
+Envoy is architected such that different types of configuration management approaches are possible.
+The approach taken in a deployment will be dependent on the needs of the implementor. Simple
+deployments are possible with a fully static configuration. More complicated deployments can
+incrementally add more complex dynamic configuration, the downside being that the implementor must
+provide one or more external REST based configuration provider APIs. This document gives an overview
+of the options currently available.
+
+* Top level configuration :ref:`reference <config>`.
+* :ref:`Reference configurations <install_ref_configs>`.
+
+Fully static
+------------
+
+In a fully static configuration, the implementor provides a set of :ref:`listeners
+<config_listeners>` (and :ref:`filter chains <config_listener_filters>`), :ref:`clusters
+<config_cluster_manager>`, and optionally :ref:`HTTP route configurations
+<config_http_conn_man_route_table>`. Dynamic host discovery is only possible via DNS based
+:ref:`service discovery <arch_overview_service_discovery>`. Configuration reloads must take place
+via the built in :ref:`hot restart <arch_overview_hot_restart>` mechanism.
+
+Though simplistic, fairly complicated deployments can be created using static configurations and
+graceful hot restarts.
+
+SDS only
+--------
+
+The :ref:`service discovery service (SDS) API <config_cluster_manager_sds>` provides a more advanced
+mechanism by which Envoy can discover members of an upstream cluster. Layered on top of a static
+configuration, SDS allows an Envoy deployment to circumvent the limitations of DNS (maximum records
+in a response, etc.) as well as consume more information used in load balancing and routing (e.g.,
```

Comment:
nit: as we do not support canary routing yet, i'd just mention weight, zone, etc.